hudi-client/hudi-spark-client/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/HoodieParquetFileFormatHelper.scala

```diff
@@ -127,6 +146,49 @@ object HoodieParquetFileFormatHelper {
       })
     }
 
+    def recursivelyApplyMultiplication(expr: Expression, columnPath: String, dataType: DataType): Expression = {
```

Comment:
I'm wondering if we can change `HoodieParquetReadSupport` and adding a read support implementation for Avro parquet reader for handling the millis interpretation, which is one layer below the current approach?  Would that incur less overhead than the projection?